theano/sandbox/cuda/basic_ops.py

```diff
@@ -4288,3 +4289,100 @@ def c_code(self, node, name, inp, out, sub):
     def c_code_cache_version(self):
         return (3,)
 gpu_eye = GpuEye(dtype='float32')
+
+
+class GpuDiagonal(GpuOp):
+    __props__ = ("offset", "axis1", "axis2")
+    default_offset = 0
+    default_axis1 = 0
+    default_axis2 = 1
+
+    def __init__(self, offset=0, axis1=0, axis2=1):
+        self.view = view
+        if self.view and not numpy_diagonal_return_view:
```

Comment:
We don't care about numpy_diagonal_return_view inside the GpuOp as it doesn't use numpy.